x-pack/plugins/security_solution/server/lib/detection_engine/signals/reason_formatter.test.ts

```diff
@@ -180,14 +191,61 @@ describe('reason_formatter', () => {
           },
         };
         expect(
-          buildReasonMessageUtil({ name, severity, mergedDoc: updatedMergedDoc })
+          buildReasonMessageUtil({ name: ruleName, severity, mergedDoc: updatedMergedDoc })
         ).toMatchInlineSnapshot(`"test event by test-user created medium alert my-rule."`);
       });
     });
     describe('when only rule is provided', () => {
       it('should return the reason message without host name or user name', () => {
-        expect(buildReasonMessageUtil({ name, severity })).toMatchInlineSnapshot(`""`);
+        expect(buildReasonMessageUtil({ name: ruleName, severity })).toMatchInlineSnapshot(`""`);
+      });
+    });
+  });
+
+  describe(`buildReasonMessageForThresholdAlert`, () => {
+    it('When thresold rule is grouped by user.name', () => {
+      const userName = 'Some User Name';
+      const thresholdMergedDoc = genThresholdMergedDoc({
+        'user.name': userName,
+      });
+      expect(
+        buildReasonMessageForThresholdAlert({
+          name: ruleName,
+          severity,
+          mergedDoc: thresholdMergedDoc,
+        })
+      ).toEqual(`event by ${userName} created ${severity} alert ${ruleName}.`);
+    });
+
+    it('When threshold rule is grouped by host.name', () => {
+      const hostName = 'Some Host Name';
+      const thresholdMergedDoc = genThresholdMergedDoc({
+        'host.name': hostName,
       });
+
+      expect(
+        buildReasonMessageForThresholdAlert({
+          name: ruleName,
+          severity,
+          mergedDoc: thresholdMergedDoc,
+        })
+      ).toEqual(`event on ${hostName} created ${severity} alert ${ruleName}.`);
+    });
+    it('When threshold rule is grouped by host.name and user.name', () => {
```

Comment:
nit: missing newline

Reply:
@andrew-goldstein , I am not completely sure that I understand this comment.

Reply:
> @andrew-goldstein , I am not completely sure that I understand this comment.

😅 "nit: Consider adding vertical whitespace above line `234`, to visually separate this `it` block from the previous test."

Reply:
I agree.. will keep this in mind.